boa_engine/src/builtins/function/mod.rs

```diff
@@ -210,6 +241,79 @@ impl Function {
             Self::Ordinary { code, .. } | Self::Generator { code, .. } => code.constructor,
         }
     }
+
+    /// Returns true if the function object is a derived constructor.
+    pub(crate) fn is_derived_constructor(&self) -> bool {
+        if let Self::Ordinary {
+            constructor_kind, ..
+        } = self
+        {
+            constructor_kind.is_derived()
+        } else {
+            false
+        }
+    }
+
+    /// Returns a reference to the function `[[HomeObject]]` slot if present.
+    pub(crate) fn get_home_object(&self) -> Option<&JsObject> {
+        if let Self::Ordinary { home_object, .. } = self {
+            home_object.as_ref()
+        } else {
+            None
+        }
+    }
+
+    ///  Sets the `[[HomeObject]]` slot if present.
+    pub(crate) fn set_home_object(&mut self, object: &JsObject) {
+        if let Self::Ordinary { home_object, .. } = self {
+            *home_object = Some(object.clone());
+        }
+    }
```

Comment:
Here, I would pass the `JsObject` by value, since we use it like that, and can avoid a clone in some cases (even if that clone is just copying a pointer value).